contracts/SpaceSplitOperation.sol

```diff
@@ -0,0 +1,277 @@
+    /* solium-disable-next-line */
+    return weilerAtherton.subjectPolygon.handledIntersectionPoints == weilerAtherton.subjectPolygon.intersectionPoints.length
+    /* solium-disable-next-line */
+        && weilerAtherton.clippingPolygon.handledIntersectionPoints == weilerAtherton.clippingPolygon.intersectionPoints.length;
+  }
+
+  function buildSubjectPolygonOutput() public {
+    require(doneStage == Stage.INTERSECT_POINTS_ADD, "doneStage should be SEGMENTS_ADD");
+    require(weilerAtherton.subjectPolygon.handledIntersectionPoints == weilerAtherton.subjectPolygon.intersectionPoints.length, "buildResultPolygon not finished");
+    require(weilerAtherton.clippingPolygon.handledIntersectionPoints == weilerAtherton.clippingPolygon.intersectionPoints.length, "buildResultPolygon not finished");
+
+    weilerAtherton.buildSubjectPolygonOutput();
+
+    doneStage = Stage.WEILER_ATHERTON_BUILD;
+  }
+
+  function processWeilerAtherton() public {
+    addIntersectedPoints();
+    buildResultPolygon();
+    buildSubjectPolygonOutput();
+  }
+
+  function convertPointsToContour(PolygonUtils.CoorsPolygon storage latLonPolygon) private returns (uint256[] geohashContour) {
+    geohashContour = new uint256[](latLonPolygon.points.length);
+
+    uint256 geohash;
+    for (uint i = 0; i < latLonPolygon.points.length; i++) {
+      geohash = splitMerge.getCachedGeohashByLatLon(latLonPolygon.points[i], 12);
+      if (geohash == 0) {
+        geohash = splitMerge.cacheLatLonToGeohash(latLonPolygon.points[i], 12);
+      }
+
+      geohashContour[i] = geohash;
+    }
+  }
+
+  function finishSubjectPolygon() public {
+    require(doneStage == Stage.WEILER_ATHERTON_BUILD, "doneStage should be WEILER_ATHERTON_BUILD");
+    require(subjectContourOutput.length == 0, "Clipping polygons already finished");
+
+    subjectContourOutput = convertPointsToContour(weilerAtherton.subjectPolygonOutput);
+    if (resultContours.length > 0) {
+      doneStage = Stage.POLYGONS_FINISH;
+    }
+  }
+
+  function finishClippingPolygons() public {
+    require(doneStage == Stage.WEILER_ATHERTON_BUILD, "doneStage should be WEILER_ATHERTON_BUILD");
+    require(resultContours.length == 0, "Clipping polygons already finished");
+
+    for (uint i = 0; i < weilerAtherton.resultPolygons.length; i++) {
+      resultContours.push(convertPointsToContour(weilerAtherton.resultPolygons[i]));
+    }
+    if (subjectContourOutput.length > 0) {
+      doneStage = Stage.POLYGONS_FINISH;
+    }
+  }
+
+  function finishAllPolygons() public {
+    finishSubjectPolygon();
+    finishClippingPolygons();
+  }
+
+  function getResultContour(uint256 contourIndex) public view returns (uint256[]) {
+    return resultContours[contourIndex];
+  }
+
+  function getFinishInfo() public view returns (uint256[] subjectContourResult, address tokenOwner, uint256 resultContoursCount) {
+    require(doneStage == Stage.POLYGONS_FINISH, "SpaceSplitOperation not finished");
```

Comment:
If this getter is not intended to be called from a state-modifying function, the require statement is useless

Reply:
This function used for getting finish info. If finish info not appeared yet - so need to tell about this. Also this function for calling from another contract - SplitMerge, and using for validate: is operation finished or not. If i will divide to two separated functions - it will be costs more gas.